standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -981,9 +981,15 @@ public static ConfVars getMetaConf(String name) {
         + " and the frequency of persisting the metrics to persistent storage. "
       ),
     REPL_METRICS_UPDATE_FREQUENCY("metastore.repl.metrics.update.frequency",
-      "hive.repl.metrics.update.frequency", 1 /*1 minute */,
+      "hive.repl.metrics.update.frequency", 1, TimeUnit.MINUTES /*1 minute */,
       "Frequency at which replication Metrics will be stored in persistent storage. "
     ),
+    REPL_METRICS_CLEANUP_FREQUENCY("metastore.repl.metrics.cleanup.frequency",
```

Comment:
REPL_METRICS_MAX_AGE set to lower value than REPL_METRICS_CLEANUP_FREQUENCY will result with retaining entries longer than REPL_METRICS_MAX_AGE set.
If REPL_METRICS_MAX_AGEis set in days, why not to fix the REPL_METRICS_CLEANUP_FREQUENCY to 1 day and not make it configurable?

Reply:
Its difficult to derive it. That is why its made configurable. Eg : if max age is set to 30 days, the clean up frequency may be a day for 1 customer, maybe 7 days for another. So this is configurable.

Reply:
This case is fine, but say customer set the REPL_METRICS_MAX_AGE as 5 days and REPL_METRICS_CLEANUP_FREQUENCY as 30 days, this will lead to confusion. REPL_METRICS_MAX_AGE is also considered and crucial to decide how much data we would end up storing at a given time and REPL_METRICS_MAX_AGE is ideally to control that. The REPL_METRICS_CLEANUP_FREQUENCY here is more influential than the REPL_METRICS_MAX_AGE which shouldn't be the case

Reply:
That they have to fix the configuration. We can't derive it.

Reply:
A similar approach is followed by other clean up tasks as well. For instance :     SCHEDULED_QUERIES_EXECUTION_MAINT_TASK_FREQUENCY("metastore.scheduled.queries.execution.maint.task.frequency",
        "hive.metastore.scheduled.queries.execution.clean.frequency", 60, TimeUnit.SECONDS,
        "Interval of scheduled query maintenance task. Which removes executions above max age;"
            + "and marks executions as timed out if the condition is met"),
    SCHEDULED_QUERIES_EXECUTION_MAX_AGE("metastore.scheduled.queries.execution.max.age",
        "hive.metastore.scheduled.queries.execution.max.age", 30 * 86400, TimeUnit.SECONDS,
        "Maximal age of a scheduled query execution entry before it is removed."),
We need to document it. I think customers will understand.

Reply:
Throw an exception is REPL_METRICS_CLEANUP_FREQUENCY is set higher than the REPL_METRICS_MAX_AGE

Reply:
Even if you throw an exception the thread running clean up will be killed. Customer can only come to know about it from the logs.

Reply:
Or, you can fix the conf description for REPL_METRICS_CLEANUP_FREQUENCY stating it's association to REPL_METRICS_MAX_AGE.

Reply:
I have added that already that it removes metrics above the max age. I will add the config name as well. That should help I think. 
REPL_METRICS_CLEANUP_FREQUENCY("metastore.repl.metrics.cleanup.frequency",
      "hive.metastore.repl.metrics.cleanup.frequency", 1, TimeUnit.DAYS,
      "Interval of scheduled metrics clean up task. Which removes metrics above max age;"),
    REPL_METRICS_MAX_AGE("metastore.repl.metrics.max.age",
      "hive.metastore.repl.metrics.max.age", 7, TimeUnit.DAYS,
      "Maximal age of a replication metrics entry before it is removed."),

Reply:
Yes, that would suffice. Mentioning the conf will help related which max age we are talking about.